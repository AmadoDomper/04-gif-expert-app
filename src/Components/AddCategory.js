import React, {useState} from "react";

export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState("Hola Mundo");

    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(cats => [...cats, inputValue]);
            setinputValue('');
        }

    }

  return (
    <form onSubmit={handleSubmit}>
      <input 
            type='text' 
            value={inputValue} 
            onChange={handleInputChange}
            />
    </form>
  );
};
