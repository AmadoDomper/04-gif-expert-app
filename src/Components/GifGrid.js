import React from 'react'

export const GifGrid = ({ category }) => {

    const getGifs = async() => {
        const url = 'http://api.giphy.com/v1/gifs/search?q=Dragon+Ball&limit=10&api_key=oevkH2leXBqB9scIAiy61szmSLgvY44X'
        const resp = await fetch( url );
        const { data } = await resp.json();

        console.log(data);

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);
    }

    getGifs();

  return (
    <div>
        <h3>
            { category }
        </h3>
    </div>
  )
}
