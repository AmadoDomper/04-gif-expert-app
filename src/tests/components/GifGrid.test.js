import { GifGrid } from "../../Components/GifGrid";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs.js');

describe('Testing in <GifGrid />', () => {  
    const categoryInput = 'Goku';

    test('should show component correctly', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={categoryInput} />);
        expect(wrapper).toMatchSnapshot();
        
    })
    
    test('should show items when call useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https//localhost/anything/thing.jpg',
            title: 'everything'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={categoryInput} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        console.log(wrapper.find('Component').length);
        // expect(wrapper.find('Component').length).toBe(gifs.length);


    })
    

})

