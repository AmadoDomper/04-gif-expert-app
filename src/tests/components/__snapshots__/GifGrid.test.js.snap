// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifGrid /> should show component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-2s"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing in <GifGrid /> should show items when call useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-2s"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <Component
      id="ABC"
      key="ABC"
      title="everything"
      url="https//localhost/anything/thing.jpg"
    />
  </div>
</Fragment>
`;
